stm32h573i_dk : Fix CAN core clock and add I2C test

marwaiehm-st opened the pull request:
Set the PLL1_Q divider to 6 give a can core clock of 80MHz to resolve fdcan_clk reception problem because M_CAN requires that the host clock "APB1" should be higher or equal to the CAN core clock "PLL1_Q".
Add necessary overlay and stm32h573i_dk in i2c_target_api test case to enable the board.

marwaiehm-st committed:
tests: drivers: i2c: i2c_target_api: add stm32h573i_dk

Add necessary overlay and stm32h573i_dk in i2c_target_api test case
to enable the board.

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

erwango commented on tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay:
```diff
@@ -0,0 +1,42 @@
+ */
+
+&i2c1 {
+       clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00200000>,
+		<&rcc STM32_SRC_PLL3_R I2C1_SEL(1)>;
+	pinctrl-0 = <&i2c1_scl_pb6 &i2c1_sda_pb7>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+	status = "okay";
```
Not required as provided by board, unless there is a specific reason to duplicate it

marwaiehm-st replied:
Yes you are right there no specific reason to duplicate, i will remove it

erwango commented on tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay:
```diff
@@ -0,0 +1,42 @@
+};
+
+&i2c2 {
+       clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00400000>,
+		<&rcc STM32_SRC_PLL3_R I2C2_SEL(1)>;
+	pinctrl-0 = <&i2c2_scl_pb10 &i2c2_sda_pb11>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+	status = "okay";
```
ditto

marwaiehm-st replied:
I will remove it

marwaiehm-st committed:
boards: stm32h573i_dk: Fix CAN core clock

Set the PLL1_Q divider to 6 give a can core clock of 80MHz to
resolve fdcan_clk reception problem because M_CAN requires that
the host clock "APB1" should be higher or equal to the CAN core
clock "PLL1_Q".

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

marwaiehm-st commented:
Related to the issue: https://github.com/zephyrproject-rtos/zephyr/issues/70557

erwango commented:
@marwaiehm-st Please update PR title

FRASTM commented on boards/st/stm32h573i_dk/stm32h573i_dk.dts:
```diff
@@ -81,7 +81,7 @@
 	mul-n = <96>;
 	div-p = <2>;
-	div-q = <2>;
+	div-q = <6>;
```
Did you check that changing the pll1_q output freq does not disturb other peripherals that have pll1_q as clock source

marwaiehm-st replied:
Yes, i checked it. The pll1_q  output frep is only used by the peripheral FDCAN  on stm32h573i_dk.

FRASTM reviewed (commented):
else LGTM

henrikbrixandersen commented:
With this change in, you should be able to add `CONFIG_TEST_ALL_BITRATES=y` for this board to `tests/drivers/can/timing/`.
